src/components/ColorPicker.tsx

```diff
@@ -42,14 +48,104 @@ const ColorSwatch: FC<{
   const dispatch = useDispatch()
   const colors = useSelector(themeColors)
   const fontSize = useSelector(state => state.fontSize)
-  const selected =
-    (color && cursorStyle?.color === color) || (backgroundColor && cursorStyle?.backgroundColor === backgroundColor)
+  const currentThoughtId = useSelector(state => !!state.cursor && head(state.cursor))
   size = size || fontSize * 1.2
 
+  /** A function that adds an alpha channel to a hex color. */
+  const addAlphaToHex = (hex: string) => {
+    if (hex.length === 7) return hex + 'ff'
+    return hex
+  }
+
+  const selected = useSelector(state => {
+    const currentThoughtValue = (!!state.cursor && getThoughtById(state, head(state.cursor))?.value) || ''
+    const themeColor = themeColors(state)
+    // Define the color and background color regex to get the current color of current thought
+    // document.execCommand('foreColor') adds the color attribute with hex and document.execCommand('backColor') adds the background-color attribute with the rgb
```

Comment:
This explains *what* it is doing, but not *why*. `document.execCommand('foreColor')` and `document.execCommand('backColor')` both work with hex or rgb, so why the difference here?

Maybe it is connected to this comment in themeColors.ts:

```
// Background colors in capacitor app needs to be in hexadecimal codes
```

There may have originally been no repercussions of specifying the background color in hex and the rest of the colors in rgb. But now we have parsing logic that is based on those colors, so that seemingly arbitrary difference is spreading outward and affecting other parts of the app. Ideally, the colors should all be the same format, and just converted to hex for capacitor as needed. In lieu of refactoring the hex background, we must at least leave a comment that clarifies the root cause of this conspicuous difference. 

Hope that makes sense :).